File: app/src/main/java/com/ckunder/recyclersample/adapter/ViewHolderFactory.java
package com.ckunder.recyclersample.adapter;

import android.content.Context;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;
import io.reactivex.annotations.NonNull;

/**
 * Instantiates a {@link RecyclerView.ViewHolder} based on the type.
 */
public interface ViewHolderFactory {

    RecyclerView.ViewHolder createViewHolder(@NonNull final ViewGroup parent, RecyclerView.RecycledViewPool viewPool);
}

File: app/src/main/java/com/ckunder/recyclersample2/genericAdapter/ViewHolderFactory.kt
package com.ckunder.recyclersample2.genericAdapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView


interface ViewHolderFactory {

    fun createViewHolder(parent: ViewGroup, viewPool: RecyclerView.RecycledViewPool? = null): RecyclerView.ViewHolder
}